src/sagemaker_sklearn_extension/preprocessing/encoders.py

```diff
@@ -423,3 +423,158 @@ def inverse_transform(self, y):
+           [2., 3.]], dtype=float32)
+
+    >>> enc.inverse_transform([[1, 0], [0, 1], [2, 3]])
+    array([['Male', 1],
+           ['Female', 2],
+           [None, None]], dtype=object)
+
+    """
+
+    def __init__(self, categories="auto", dtype=np.float32):
+        super(RobustOrdinalEncoder, self).__init__(categories, dtype)
+        self.categories = categories
+        self.dtype = dtype
+
+    def fit(self, X, y=None):
+        """Fit the OrdinalEncoder to X.
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to determine the categories of each feature.
+
+        Returns
+        -------
+        self
+
+        """
+        # base classes uses _categories to deal with deprecations in
+        # can be removed once deprecations are removed
+        self._categories = self.categories
+        self._fit(X, handle_unknown="unknown")
+        return self
+
+    def transform(self, X):
+        """Transform X to ordinal codes.
+
+        Parameters
+        ----------
+        X : array-like, shape [n_samples, n_features]
+            The data to encode.
+
+        Returns
+        -------
+        X_out : sparse matrix or a 2-d array
+            Transformed input.
+
+        """
+        X_int, X_mask = self._transform(X, handle_unknown="unknown")
+        # assign the unknowns an integer indicating they are unknown. The largest integer is always reserved for
+        # unknowns
+        for col in range(X_int.shape[1]):
+            mask = X_mask[:, col]
+            X_int[~mask, col] = self.categories_[col].shape[0]
+
+        return X_int.astype(self.dtype, copy=False)
+
+    def inverse_transform(self, X):
+        """Convert the data back to the original representation.
+        In slots where the encoding is that of an unrecognised category, the output of the inverse transform is np.nan
+        for float or complex arrays, and None otherwise
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, shape [n_samples, n_encoded_features]
+            The transformed data.
+
+        Returns
+        -------
+        X_tr : array-like, shape [n_samples, n_features]
+            Inverse transformed array.
+
+        """
+        check_is_fitted(self, "categories_")
+        X = check_array(X, accept_sparse="csr")
+
+        n_samples, _ = X.shape
+        n_features = len(self.categories_)
+
+        # validate shape of passed X
+        msg = "Shape of the passed X data is not correct. Expected {0} " "columns, got {1}."
+        if X.shape[1] != n_features:
+            raise ValueError(msg.format(n_features, X.shape[1]))
+
+        # create resulting array of appropriate dtype
+        dt = np.find_common_type([cat.dtype for cat in self.categories_], [])
+        X_tr = np.empty((n_samples, n_features), dtype=dt)
+
+        found_unknown = {}
+        for i in range(n_features):
+            labels = X[:, i].astype("int64", copy=False)
```

Comment:
This line will error if the input X cannot be cast into int64, you can specify the dtype in `check_array` on L548

Reply:
check_array has dtype="numeric" by default. Any numeric should be possible to convert to integers. I'll add this parameter explicitly.